From 282b1c9ea58cc0f2337a72912808505e5f540d5a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20Men=C5=A1=C3=ADk?= <pemensik@redhat.com>
Date: Sat, 28 Jan 2023 11:32:05 +0100
Subject: [PATCH] Minimal change to compile with BIND 9.18.11
 
DSCP codes are not working and their support were removed from BIND9. Do
not require them to be present.
---
 src/fwd.c | 4 ++++
 1 file changed, 4 insertions(+)
 
diff --git a/src/fwd.c b/src/fwd.c
index 24f6e53..10ec848 100644
--- a/src/fwd.c
+++ b/src/fwd.c
@@ -198,7 +198,9 @@ fwd_print_list_buff(isc_mem_t *mctx, dns_forwarders_t *fwdrs,
 	     INSIST((fwdr_int == NULL) == (fwdr_cfg == NULL)), fwdr_int != NULL;
 	     fwdr_int = ISC_LIST_NEXT(fwdr_int, link), fwdr_cfg = cfg_list_next(fwdr_cfg)) {
 		fwdr_cfg->obj->value.sockaddrdscp.sockaddr = fwdr_int->addr;
+#if LIBDNS_VERSION_MAJOR < 1811
 		fwdr_cfg->obj->value.sockaddrdscp.dscp = fwdr_int->dscp;
+#endif
 	}
 	cfg_print(faddresses, buffer_append_str, &tmp_buf);
 
@@ -281,7 +283,9 @@ fwd_parse_str(const char *fwdrs_str, isc_mem_t *mctx,
 			isc_sockaddr_setport(&addr, port);
 		fwdr = isc_mem_get(mctx, sizeof(*(fwdr)));
 		fwdr->addr = addr;
+#if LIBDNS_VERSION_MAJOR < 1811
 		fwdr->dscp = cfg_obj_getdscp(fwdr_cfg);
+#endif
 		ISC_LINK_INIT(fwdr, link);
 		ISC_LIST_APPEND(*fwdrs, fwdr, link);
 	}
-- 
2.39.1